src/MerchantCenter/AccountService.php

```diff
@@ -623,4 +632,45 @@ public function update_wpcom_api_authorization( string $status, string $nonce ):
 	public function delete_wpcom_api_auth_nonce(): bool {
 		return $this->options->delete( OptionsInterface::GOOGLE_WPCOM_AUTH_NONCE );
 	}
+
+	/**
+	 * Deletes the transient storing the WPCOM Status data.
+	 */
+	public function delete_wpcom_api_status_transient(): void {
+		$transients = $this->container->get( TransientsInterface::class );
+		$transients->delete( TransientsInterface::WPCOM_API_STATUS );
+	}
+
+	/**
+	 * Check if the WPCOM API Status is healthy by doing a request to /wc/partners/google/remote-site-status endpoint in WPCOM.
+	 *
+	 * @return bool True when the status is healthy, false otherwise.
+	 */
+	public function is_wpcom_api_status_healthy() {
+		/** @var TransientsInterface $transients */
+		$transients = $this->container->get( TransientsInterface::class );
+		$status     = $transients->get( TransientsInterface::WPCOM_API_STATUS );
+
+		if ( ! $status ) {
+
+			$integration_status_args = [
+				'method'  => 'GET',
+				'timeout' => 30,
+				'url'     => 'https://public-api.wordpress.com/wpcom/v2/sites/' . Jetpack_Options::get_option( 'id' ) . '/wc/partners/google/remote-site-status',
+				'user_id' => get_current_user_id(),
+			];
+
+			$integration_remote_request_response = Client::remote_request( $integration_status_args, null );
+
+			if ( is_wp_error( $integration_remote_request_response ) ) {
+				$this->delete_wpcom_api_status_transient();
+				return false;
+			}
+
+			$status = json_decode( wp_remote_retrieve_body( $integration_remote_request_response ), true ) ?? [];
```

Comment:
If `json_decode` returns `null`, we currently set it to an empty array, which means we'll keep trying to get the status repeatedly. Instead, shouldn't we set the transient to something like `['is_healthy' => false]` to avoid querying the endpoint?